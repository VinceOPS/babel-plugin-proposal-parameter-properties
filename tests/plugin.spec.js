const babel = require('babel-core');
const plugin = require('../').default;
const syntaxDecorators = require('babel-plugin-syntax-decorators');

describe('Class declaration syntax', () => {
    it('does not transform when decorator is not valid', () => {
        const example = '@baddecorator class DoNotCare { constructor(prop1) {} }';
        const {code} = babel.transform(example, {plugins: [syntaxDecorators, plugin]});
        expect(code).toMatchSnapshot();
    });

    it('does transform when decorator is valid', () => {
        const example = '@paramProperties class DoCare { constructor(prop1) {} }';
        const {code} = babel.transform(example, {plugins: [syntaxDecorators, plugin]});
        expect(code).toMatchSnapshot();
    });
});