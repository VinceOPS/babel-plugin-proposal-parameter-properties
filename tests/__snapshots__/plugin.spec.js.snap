// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decorated classes only:  does not transform when decorator is not valid 1`] = `
"@baddecorator
class DoNotCare {
  constructor(prop1) {}
}"
`;

exports[`Decorated classes only:  does not transform when no decorator are set 1`] = `
"class DoNotCare {
  constructor(prop1) {}
}"
`;

exports[`Decorated classes only:  does transform when decorator is valid 1`] = `
"class DoCare {
  constructor(prop1) {
    this.prop1 = prop1;
  }
}"
`;

exports[`Transformation:  always puts assignments after a call to super() 1 1`] = `
"class DoCare extends Imaginary {
  constructor(prop1, prop2) {
    super();
    this.prop1 = prop1;
    this.prop2 = prop2;
  }
}"
`;

exports[`Transformation:  always puts assignments after a call to super() 2 1`] = `
"class DoCare extends Imaginary {
  constructor(prop1, prop2) {
    console.log(\\"ok\\");
    super();
    this.prop1 = prop1;
    this.prop2 = prop2;
  }
}"
`;
